staging/src/k8s.io/pod-security-admission/policy/check_seccomp_restricted.go

```diff
@@ -0,0 +1,141 @@
+		Level: api.LevelRestricted,
+		Versions: []VersionedCheck{
+			{
+				MinimumVersion: api.MajorMinorVersion(1, 0),
+				CheckPod:       seccomp_1_0_restricted,
+			},
+			{
+				MinimumVersion: api.MajorMinorVersion(1, 19),
+				CheckPod:       seccomp_1_19_restricted,
+			},
+		},
+	}
+}
+
+// seccomp_1_0_restricted checks restricted policy on seccomp alpha annotation
+func seccomp_1_0_restricted(podMetadata *metav1.ObjectMeta, podSpec *corev1.PodSpec) CheckResult {
+	forbidden := sets.NewString()
+	podSeccompSet := false
+
+	if val, ok := podMetadata.Annotations[annotationKeyPod]; ok {
+		if !validSeccompAnnotation(val) {
+			podAnnotationField := field.NewPath("metadata").Child("annotations", annotationKeyPod)
+			forbidden.Insert(fieldValue(podAnnotationField, val))
+		} else {
+			podSeccompSet = true
+		}
+	}
+
+	visitContainersWithPath(podSpec, field.NewPath("spec"), func(c *corev1.Container, path *field.Path) {
+		annotation := annotationKeyContainerPrefix + c.Name
+		if val, ok := podMetadata.Annotations[annotation]; ok {
+			if val == corev1.SeccompProfileNameUnconfined {
+				containerAnnotationField := field.NewPath("metadata").
+					Child("annotations", annotation)
+				forbidden.Insert(fieldValue(containerAnnotationField, val))
+			}
+		} else if !podSeccompSet {
+			containerAnnotationField := field.NewPath("metadata").
+				Child("annotations", annotation)
+			forbidden.Insert(fieldValueRequired(containerAnnotationField))
+		}
+	})
+
+	if len(forbidden) > 0 {
+		return CheckResult{
+			Allowed:         false,
+			ForbiddenReason: "seccomp profile",
+			ForbiddenDetail: strings.Join(forbidden.List(), ", "),
+		}
+	}
+
+	return CheckResult{Allowed: true}
+}
+
+// seccomp_1_19_restricted checks restricted policy on securityContext.seccompProfile field
+func seccomp_1_19_restricted(podMetadata *metav1.ObjectMeta, podSpec *corev1.PodSpec) CheckResult {
+	forbidden := sets.NewString()
+	podSeccompField := field.NewPath("spec").Child("securityContext", "seccompProfile", "type")
+	podSeccompSet := false
+
+	if podSpec.SecurityContext != nil {
+		if podSpec.SecurityContext.SeccompProfile != nil {
+			if podSpec.SecurityContext.SeccompProfile.Type != "" {
+				seccompType := podSpec.SecurityContext.SeccompProfile.Type
+				if !validSeccomp(podSpec.SecurityContext.SeccompProfile.Type) {
+					forbidden.Insert(fieldValue(podSeccompField, string(seccompType)))
+				} else {
+					podSeccompSet = true
+				}
+			}
+		}
+	}
+
+	visitContainersWithPath(podSpec, field.NewPath("spec"), func(c *corev1.Container, path *field.Path) {
+		if c.SecurityContext != nil {
+			if c.SecurityContext.SeccompProfile != nil {
+				if c.SecurityContext.SeccompProfile.Type != "" {
+					seccompType := c.SecurityContext.SeccompProfile.Type
+					if !validSeccomp(seccompType) {
+						containerSeccompField := path.Child("securityContext", "seccompProfile", "type")
+						forbidden.Insert(fieldValue(containerSeccompField, string(seccompType)))
+					}
```

Comment:
would be simpler to return if the container explicitly sets to a valid type, and collapse all the various ways the container might not explicitly set SeccompProfile to a single check:

```go
if c.SecurityContext != nil && c.SecurityContext.SeccompProfile != nil {
	seccompType := c.SecurityContext.SeccompProfile.Type
	if !validSeccomp(seccompType) {
		containerSeccompField := path.Child("securityContext", "seccompProfile", "type")
		forbidden.Insert(fieldValue(containerSeccompField, string(seccompType)))
	}
	return
}

if !podSeccompSet {
	containerSeccompField := path.Child("securityContext", "seccompProfile", "type")
	forbidden.Insert(fieldValueRequired(containerSeccompField))
}
```